### ISSUE TITLE
[no current rule] A new rule proposal to check if a signal is being called in a template expression.

### ISSUE THREAD

**Issue by michaeljota** [angular-eslint/angular-eslint#1425]
Function calls in template expressions are not something most Angular devs are used to, and we even have a rule to disable those. With the introduction of signals, this will change, but we may forget to call the signals to get their value. Of course, sometimes this will be evident, but sometimes we will have weirds false positives. Anyway, I think we could introduce a new rule, that checks signals that are not being called in template expressions. 

I think we should narrow this just to check template expressions, as we could pass signals as inputs, and we could use the signal in different ways in the component, but in template expressions most of the time (like 99.9% of the time) we will want the value of the signal, and not the signal itself. 

```JSON
{
  "rules": {
    "@angular-eslint/template/<rule>": ["<setting>"]
  }
}
```

```HTML
<ng-container *ngIf="!isLoading; else loader"> // always will be false
  <pre>
    {{ users() | json }}
  </pre>

  <pre *ngIf="hasError"> // always will be true
    Query with error
  </pre>
</ng-container>

<ng-template #loader> Loading... </ng-template>
```
```ts
const isLoading = signal(false)
const hasError = signal(true)
```

**Versions**

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin-template`  | `X.Y.Z` |
| `@angular-eslint/template-parser`         | `X.Y.Z` |
| `@typescript-eslint/parser`               | `X.Y.Z` |
| `ESLint`                                  | `X.Y.Z` |
| `node`                                    | `X.Y.Z` |

```sh
# Please run `npx ng version` in your project and paste the full output here:
Angular CLI: 16.0.4
Node: 16.15.0
Package Manager: npm 8.5.5
OS: win32 x64

Angular: 16.0.4
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1600.4
@angular-devkit/build-angular   16.0.4
@angular-devkit/core            16.0.4
@angular-devkit/schematics      16.0.4
@schematics/angular             16.0.4
ng-packagr                      16.0.1
rxjs                            7.8.1
typescript                      4.9.5
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by JeanMeche** [comment]
This check is provided by an extended diagnostic in v17 ! 

Cf angular/angular#49660

**Comment by JamesHenry** [comment]
That's great news! Thank you @JeanMeche 

FYI There is currently no way for us to implement such rules in angular-eslint because we do not have access to the template type checker, only the angular compiler itself does.

**Comment by emeryao** [comment]
A little suggestion here 🙋‍♂️

How about make the option `allowList` support ***regex*** ?

We can define a common regex like some string ending with `Signal` in the `allowList` and then we can name all the signal variables to be end with `Signal` like *`isLoadingSignal`* to follow the regex so the linter should not warning about the `{{ isLoadingSignal() }}` in html template any more

**Comment by JamesHenry** [comment]
@emeryao That is not related to this issue, this issue was about making sure a signal is always called, not false positives around calling functions in templates.

If you want to submit a PR for an enhancement to the no-call-expression rule which adds regexp powered exceptions then please do

**Comment by maxime1992** [comment]
From @JeanMeche 

> This check is provided by an extended diagnostic in v17 ! Cf https://github.com/angular/angular/pull/49660

Just FYI for anyone else coming through this issue, I've opened a follow up here https://github.com/angular/angular/issues/57729 as it doesn't show a warning when passing a signal to a pipe which I think is unexpected (and did cost me a good debugging session :upside_down_face: !)
